cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/sidebar.spec.js

```diff
@@ -223,6 +315,73 @@ export const runSideBarTests = () => {
         it('handles panel collapse/expand correctly', () => {
           checkSidebarPanelBehavior();
         });
+
+        it('fields should have top values', () => {
+          const aggregatableFieldsToTest = [sidebarFields.aggregatableFields.unnested[0]];
+          const nestedFieldsToTest = [sidebarFields.aggregatableFields.nested[0]];
+
+          aggregatableFieldsToTest.forEach((aggregatableField) => {
+            verifyFieldShowDetailsShowsTopValuesAndViewVisualization(
+              config,
+              aggregatableField,
+              true
+            );
+          });
+
+          cy.navigateToWorkSpaceSpecificPage({
+            url: BASE_PATH,
+            workspaceName: workspaceName,
+            page: 'discover',
+            isEnhancement: true,
+          });
+          cy.getElementByTestId('discoverNewButton').click();
+          // Setting the dataset and query language again to ensure the date picker is not missing
+          cy.setDataset(config.dataset, DATASOURCE_NAME, config.datasetType);
+          cy.setQueryLanguage(config.language);
+
+          nestedFieldsToTest.forEach((nestedField) => {
+            verifyFieldShowDetailsShowsTopValuesAndViewVisualization(config, nestedField, false);
+          });
+        });
+
+        it('fields should be filtered by type', () => {
+          cy.getElementByTestId('toggleFieldFilterButton').click();
+
+          sideBar.selectFilterVerifySidebarFieldsVisibleAndActiveFiltersNumber(
+            'aggregatable',
+            sidebarFields.aggregatableFields.unnested.concat(
+              sidebarFields.aggregatableFields.nested
+            )
+          );
+
+          // TODO: Index dataset type does not support searchable fields.
+          // https://github.com/opensearch-project/OpenSearch-Dashboards/issues/9381
+          if (config.datasetType !== 'INDEXES') {
+            sideBar.selectFilterVerifySidebarFieldsVisibleAndActiveFiltersNumber(
+              'searchable',
+              sidebarFields.searchableFields
+            );
+          }
+
+          cy.getElementByTestId('missingSwitch').click();
+          sidebarFields.missingFields.forEach((fieldName) => {
```

Comment:
Currently hide missing values only work for DQL and Lucene https://github.com/opensearch-project/OpenSearch-Dashboards/issues/9342

Though the tests passed, it is due to PPL and SQL will display all mapped fields including missing values fields in the side panel. It is a false positive. Will submit a PR to update this part.